plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,176 @@
+package supervisor
+
+import (
+	"net/url"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/kolo/xmlrpc"
+)
+
+type Supervisor struct {
+	Log telegraf.Logger `toml:"-"`
+
+	Server         string   `toml:"url"`
+	PidGather      bool     `toml:"gather_pid"`
+	ExitCodeGather bool     `toml:"gather_exit_code"`
+	UseIdentTag    bool     `toml:"use_identification_tag"`
+	FieldsInc      []string `toml:"fields_includes"`
+	FieldsExc      []string `toml:"fields_excludes"`
+
+	status supervisorInfo
+
+	rpcClient *xmlrpc.Client
+}
+
+type processInfo struct {
+	Name          string `xmlrpc:"name"`
+	Group         string `xmlrpc:"group"`
+	Description   string `xmlrpc:"description"`
+	Start         int32  `xmlrpc:"start"`
+	Stop          int32  `xmlrpc:"stop"`
+	Now           int32  `xmlrpc:"now"`
+	State         int16  `xmlrpc:"state"`
+	Statename     string `xmlrpc:"statename"`
+	StdoutLogfile string `xmlrpc:"stdout_logfile"`
+	StderrLogfile string `xmlrpc:"stderr_logfile"`
+	SpawnErr      string `xmlrpc:"spawnerr"`
+	ExitStatus    int8   `xmlrpc:"exitstatus"`
+	Pid           int32  `xmlrpc:"pid"`
+}
+
+type supervisorInfo struct {
+	StateCode int8   `xmlrpc:"statecode"`
+	StateName string `xmlrpc:"statename"`
+	Ident     string
+}
+
+const sampleConfig = `
+  ## Url of supervisor's XML-RPC endpoint if basic auth enabled in supervisor http server,
+  ## than you have to add credentials to url (ex. http://login:pass@localhost:9001/RPC2)
+  # url="http://localhost:9001/RPC2"
+  ## Use supervisor identification string as server tag
+  use_identification_tag = false
+  ## Gather PID of running processes
+  gather_pid = false
+  ## Gather exit codes of processes
+  gather_exit_code = false
```

Comment:
Please document `field_includes` and `field_excludes` here as well as in the `README.md` file.

Reply:
I will do this when this logic will be fully implemented.